src/target/riscv/riscv-013.c

```diff
@@ -2557,9 +2563,17 @@ static bool riscv013_is_halted(struct target *target)
 	if (dmstatus_read(target, &dmstatus, true) != ERROR_OK)
 		return false;
 	if (get_field(dmstatus, DMI_DMSTATUS_ANYUNAVAIL))
-		LOG_ERROR("hart %d is unavailiable", riscv_current_hartid(target));
+		LOG_ERROR("hart %d is unavailable", riscv_current_hartid(target));
 	if (get_field(dmstatus, DMI_DMSTATUS_ANYNONEXISTENT))
 		LOG_ERROR("hart %d doesn't exist", riscv_current_hartid(target));
+	if (get_field(dmstatus, DMI_DMSTATUS_ANYHAVERESET)) {
+		int hartid = riscv_current_hartid(target);
+		LOG_INFO("hart %d unexpectedly reset!", hartid);
```

Comment:
there are a lot of changes in the PR... is this basically the only functional change?

Reply:
There are a few, all to do with looking for the havereset bits.
Also, we now check dmstatus in poll() while the target is halted, which previously didn't happen.